File: FinishProject/ChineseAuctionAngular/ChineseAuctionAngular/src/app/models/PackageDto.ts
export class Package {
    idPackage: number = 0;
    name: string = '';
    price: number = 0;
    amount_Regular: number = 0;
    amount_Premium: number = 0;
    description?: string;

    constructor(init?: Partial<Package>) {
        Object.assign(this, init);
    }
}

export class PackageDto {
    name: string = '';
    description: string = '';
    price: number = 0;
    amount_Regular: number = 0;
    amount_Premium: number = 0;

    constructor(init?: Partial<PackageDto>) {
        Object.assign(this, init);
    }
}

File: ChineseAuction.Api.Angular/server/ChineseAuctionAPI/DTOs/PackageDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ChineseAuctionAPI.DTOs
{
    public class PackageDTO
    {
        [Required]
        public int IdPackage { get; set; }
        public int AmountRegular { get; set; }
        public int? AmountPremium { get; set; }
        [Required,Range(0,int.MaxValue)]
        public int Price { get; set; }
        [Required]
        public string Name { get; set; }
        public string? Description { get; set; }
    }

    public class PackageCreateDTO
    {
        [Required]
        public int AmountRegular { get; set; }
        public int? AmountPremium { get; set; }
        [Required, Range(0, int.MaxValue)]
        public int Price { get; set; }
        [Required]
        public string Name { get; set; }
        public string? Description { get; set; }
    }
    public class OrdersPackageDTO
    {
        public int IdPackage { get; set; } // שימי לב לאותיות קטנות כדי שיתאים ל-Angular שלך
        public int Quantity { get; set; }
    }
}